### DISCUSSION TITLE
Error in cmake in examples

### DISCUSSION THREAD

**Discussion by IlyaKodua** [AdaptiveCpp/AdaptiveCpp#694] [Q&A]
Hello, when i run cmake hipSYCL/examples/CMakeLists.txt

in file /usr/local/lib/cmake/hipSYCL/hipsycl-config.cmake

i have cmake first error:
hipsycl-config.cmake:54: error: list does not recognize sub-command JOIN CMakeLists.txt:5 (find_package)
second error:
hipsycl-config.cmake:91: error: More than one hipSYCL target platform is available. Must specify desired target(s) [and associated device(s)] via HIPSYCL_TARGETS, in the format HIPSYCL_TARGETS=<platform1>[:arch1[,arch2][..,archN]][;<platform2>[:arch1][...]][..;<platformN>] where platforms are one of . CMakeLists.txt:5 (find_package)

I have two platforms in clinfo:
opencl 2.1 for intel and opencl 3.0 for cuda

**Comment by illuhad** [comment]
Hi!

as it instructs you, you need to tell it which target to compile for. For example, `-DHIPSYCL_TARGETS="omp;cuda:sm_61" ` would compile for CPU and a Pascal-era NVIDIA GPU. `$HIPSYCL_INSTALL_PREFIX/bin/syclcc --help` will provide more information for the `--hipsycl-targets` specification and which options are possible.

hipSYCL does not rely on OpenCL, so clinfo output is irrelevant.

**Reply by IlyaKodua** [reply]
Thank you for the answer!! 

But i have flags:

HIPSYCL_TARGETS = cuda:sm_61
HIPSYCL_PLATFORM = cuda
HIPSYCL_GPU_ARCH = sm_61

And i have first error in: list(JOIN HIPSYCL_PLATFORM_OPTIONS " | " HIPSYCL_PLATFORMS_STRING)

And comand return this result: 
$ ./syclcc --hipsycl-targets='cuda' --hipsycl-cuda-path='/usr/local/cuda-11.3'
Traceback (most recent call last):
  File "./syclcc", line 1087, in <module>
    c = compiler(config)
  File "./syclcc", line 931, in __init__
    self._backends.append(cuda_invocation(config))
  File "./syclcc", line 786, in __init__
    raise OptionNotSet("No CUDA targets specified")
__main__.OptionNotSet: No CUDA targets specified

**Reply by illuhad** [reply]
Don't provide `HIPSYCL_TARGETS` simultaneously with `HIPSYCL_PLATFORM` and `HIPSYCL_GPU_ARCH`. `HIPSYCL_PLATFORM` and `HIPSYCL_GPU_ARCH` are deprecated and superseded by `HIPSYCL_TARGETS`.

> And i have first error in: list(JOIN HIPSYCL_PLATFORM_OPTIONS " | " HIPSYCL_PLATFORMS_STRING)

Please provide your full cmake invocation. Both for the example application as well as the commands that you have used to built hipSYCL.

> $ ./syclcc --hipsycl-targets='cuda' --hipsycl-cuda-path='/usr/local/cuda-11.3'

As the error suggests, you need to provide a target architecture for CUDA. E.g. `syclcc --hipsycl-targets="cuda:sm_61"`.

**Reply by IlyaKodua** [reply]
Thanks you, but now i have this error, for some reason cuda path is'n like indicated by me : 
> $ ./syclcc --hipsycl-targets='cuda:sm_61' --hipsycl-cuda-path='/usr/local/cuda-11.3' -v
clang version 10.0.0-4ubuntu1~18.04.2 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-10/bin
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Candidate multilib: .;@m64
Selected multilib: .;@m64
Found CUDA installation: /usr/local/cuda, version 7.0
 "/usr/bin/ld" -z relro --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/crt1.o /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/8/crtbegin.o -L/usr/local/bin/../lib/ -L/usr/local/cuda-11.3/lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/8 -L/usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/8/../../.. -L/usr/lib/llvm-10/bin/../lib -L/lib -L/usr/lib -lhipSYCL-rt -rpath=/usr/local/bin/../lib/ -rpath=/usr/local/cuda-11.3/lib64 -lcudart -lboost_context -lboost_fiber -lomp -rpath=/usr/lib/x86_64-linux-gnu -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/8/crtend.o /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/crtn.o
/usr/bin/ld: cannot open output file a.out: Permission denied
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Reply by illuhad** [reply]
clang is just saying that for whatever reason, it cannot write an output file to the current directory (are you in a directory where you don't hav write permissions?). This does not look like a hipSYCL issue.

**Reply by IlyaKodua** [reply]
okey, thank you very much for help!!
